lib/processor/arm/cpu.h

```diff
@@ -14,4 +14,32 @@
 
 #define metal_cpu_yield()
 
+static inline void metal_processor_io_write64(atomic_ullong *ptr,
```

Comment:
FYI, this code is a copy past from microblaze processor. 
FYI, this code is a copy past from microblaze processor. 
I had hesitated ,  but i did not find something specify that the access should be atomic. i would let it without protection, as it could be added later if needed. 
Please tell me your preference

Reply:
The MicroBlaze code will need be updated. I would prefer atomic as this operation is to write 64bit data, if it is not atomic, you cannot guarantee it writes 64bit successfully.

Reply:
Lock protection does not solve the issue. lock just protect from a concurent access in 64 bits, but a 32 bit access can be done in parallel. 
looking deeper in code, an alternative solution would be to not implement the 64 access if NO_ATOMIC_64_SUPPORT is set.  Means in io.h :
-	else if (ptr && sizeof(atomic_ullong) == width)
#ifndef NO_ATOMIC_64_SUPPORT
+	else if (ptr && sizeof(atomic_ullong) == width)
		atomic_store_explicit((atomic_ullong *)ptr, value, order);
- #else
-		metal_processor_io_write64((atomic_ullong *)ptr, value, order);
#endif
	else
		metal_assert (0);
As openamp doesn't perform 64 bit access, this seems reasonnable that application directly handles the protection of a 64 bit access, if needed for platform that doers not support  atomic 64 bit access ( for instance microblaze)

Reply:
You are right, even if we have lock inside the write/read64 it will not solve race complete as it is possible that there will be write/read32 or others at the same time, even though this is more like an application issue. for now, let's just assert if for write/read64 if NO 64bit  support.